Terminal.Gui/Views/Scroll/Scroll.cs

```diff
@@ -0,0 +1,272 @@
+﻿#nullable enable
+
+using System.ComponentModel;
+
+namespace Terminal.Gui;
+
+/// <summary>
+///     Indicates the position and size of scrollable content. The indicator can be dragged with the mouse. Can be
+///     oriented either vertically or horizontally. Used within a <see cref="ScrollBar"/>.
+/// </summary>
+/// <remarks>
+///     <para>
+///         By default, this view cannot be focused and does not support keyboard.
+///     </para>
+/// </remarks>
+public class Scroll : View
+{
+    /// <inheritdoc/>
+    public Scroll ()
+    {
+        _slider = new ();
+        Add (_slider);
+
+        WantContinuousButtonPressed = true;
+        CanFocus = false;
+        Orientation = Orientation.Vertical;
+
+        Width = Dim.Auto (DimAutoStyle.Content, 1);
+        Height = Dim.Auto (DimAutoStyle.Content, 1);
+    }
+
+    internal readonly ScrollSlider _slider;
+    private Orientation _orientation;
+    private int _position;
+    private int _size;
+    private bool _keepContentInAllViewport = true;
+
+    /// <inheritdoc/>
+    public override void EndInit ()
+    {
+        base.EndInit ();
+
+        AdjustScroll ();
+    }
+
+    /// <summary>Get or sets if the view-port is kept in all visible area of this <see cref="Scroll"/></summary>
+    public bool KeepContentInAllViewport
```

Comment:
I'm trying to understand this.

In v1 there was a similarly named construct that was super confusing to me. 

In the ContentScrolling work I did, I thought I addressed the user scenario that drove this with 

```cs
/// <summary>
///     If set, <see cref="View.Viewport"/><c>.X</c> can be set values greater than <see cref="View.GetContentSize ()"/>
///     <c>.Width</c> enabling scrolling beyond the right
///     of the content area.
/// </summary>
/// <remarks>
///     <para>
///         When not set, <see cref="View.Viewport"/><c>.X</c> is constrained to <see cref="View.GetContentSize ()"/>
///         <c>.Width - 1</c>.
///         This means the last column of the content will remain visible even if there is an attempt to scroll the
///         Viewport past the last column.
///     </para>
///     <para>
///         The practical effect of this is that the last column of the content will always be visible.
///     </para>
/// </remarks>
AllowXGreaterThanContentWidth = 4,

/// <summary>
///     If set, <see cref="View.Viewport"/><c>.Y</c> can be set values greater than <see cref="View.GetContentSize ()"/>
///     <c>.Height</c> enabling scrolling beyond the right
///     of the content area.
/// </summary>
/// <remarks>
///     <para>
///         When not set, <see cref="View.Viewport"/><c>.Y</c> is constrained to <see cref="View.GetContentSize ()"/>
///         <c>.Height - 1</c>.
///         This means the last row of the content will remain visible even if there is an attempt to scroll the Viewport
///         past the last row.
///     </para>
///     <para>
///         The practical effect of this is that the last row of the content will always be visible.
///     </para>
/// </remarks>
AllowYGreaterThanContentHeight = 8,
```

Specifically:

>  The practical effect of this is that the last column of the content will always be visible.

Can you verify we are talking about the same end-user scenario:

"As a user, when using a ScrollBar to scroll through content, the last row/column of the content never scrolls beyond the visible area."

Reply:
> Specifically:
> 
> > The practical effect of this is that the last column of the content will always be visible.
> 
Correct.

> Can you verify we are talking about the same end-user scenario:
> 
I think so.

> "As a user, when using a ScrollBar to scroll through content, the last row/column of the content never scrolls beyond the visible area."

Correct. But imagine if a user want to use a scroll bar without a `View`. But I may not being dealing with this right. What do you propose?

Reply:
> Correct. 

> But imagine if a user want to use a scroll bar without a View. But I may not being dealing with this right. What do you propose?

If someone wants to use a scroll bar as a standalone View to control another View, and they want to ensure some content never gets scrolled out of visibility, they can code that up as needed. There's no need for ScrollBar to have that functionality.

I propose we don't have a `KeepContentInViewport`-like property. `AllowYGreaterThanContentHeight` should work for most mainline scenarios.